### PR TITLE
Organisations page

### PR DESCRIPTION
**Pull request by tinashechiraya**
[Screencast from 31-08-23 13:34:56.webm](https://github.com/kartoza/sawps/assets/70011086/2df8b381-5bae-4e8a-8ad3-27b7c8f544be)

Descriptions:
the switch organisation has been removed from the dropdown in the navigation from the header to this page
on this page user can see the list of organisations they have access to 
the user can choose the organisation they wish to set active 
clicking on an organisation that is currently active will navigate to the organisation users page

### CODE REVIEW

**Comment by dimasciput** [comment]
@tinashechiraya selected organisatation need to be stored in database not in session anymore

**Comment by tinashechiraya** [comment]
@dimasciput Yes I made a comment on that , I asked if I should proceed with making another ticket on that because changing the the active organisation from being stored in the session to the db will also affect other parts of the code so I will have to update them as well

**Comment by dimasciput** [comment]
@tinashechiraya thanks, can you add it here please

**Comment by tinashechiraya** [comment]
@dimasciput Alright no problem

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 29 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 21 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 21 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 47 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 25 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 24 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 21 min

**Comment by tinashechiraya** [comment]
@dimasciput I have now added a new field on the user profile that will store their current/active organisation .
I have also replaced all the places in the code that were using the current organisation from the session , it will now use the new utility function to fetch the current organisation from the user profile.
I have also update a few tests so they don't break

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 41 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 22 min

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 23 min

File: django_project/frontend/api_views/map.py [hunk 1]
```diff
@@ -250,22 +253,26 @@ class FindPropertyByCoord(APIView):
...
     permission_classes = [IsAuthenticated]
 
-    def get(self, *args, **kwargs):
+    def get(self, request, *args, **kwargs):
         lat = self.request.GET.get('lat', 0)
         lng = self.request.GET.get('lng', 0)
         point = Point(float(lng), float(lat), srid=4326)
+        current_organisation_id = get_current_organisation_id(
+            request.user
+        ) or 0
```
Comment (dimasciput) [inline django_project/frontend/api_views/map.py:262]: Could you remove this? Just let it returns None.

File: django_project/stakeholder/test_view.py [hunk 2]
```diff
@@ -673,7 +693,7 @@ def test_notifications_page(self):
         }
         request = self.factory.post(url, data)
         request.user = self.user
-        request.session = {CURRENT_ORGANISATION_ID_KEY: self.organisation.id}
+        # request.session = {CURRENT_ORGANISATION_ID_KEY: self.organisation.id}
```
Comment (dimasciput) [inline django_project/stakeholder/test_view.py:696]: Please just remove all commented codes.

**Review by dimasciput — CHANGES_REQUESTED** [review]

**Review by dimasciput — APPROVED** [review]

**Pull request merged**